src/Game.java

```diff
@@ -1,278 +1,267 @@
-        while (true) {
-            boolean isValid = true;
-            if (scanner.hasNext()) {
-                String string = scanner.next();
-
-                for (int i = 0; i < string.length(); i++) {
-                    if (!Character.isLetter(string.charAt(i))) {
-                        isValid = false;
-                        System.out.println("Not a valid name, please use only letters.");
-                        break;
+        /**
+         * Method to handle input
+         *
+         * @return a String
+         */
+        public String getNextString() {
+            while (true) {
+                boolean isValid = true;
+                if (scanner.hasNext()) {
+                    String string = scanner.next();
+
+                    for (int i = 0; i < string.length(); i++) {
+                        if (!Character.isLetter(string.charAt(i))) {
+                            isValid = false;
+                            System.out.println("Not a valid name, please use only letters.");
+                            break;
+                        }
+                    }
+                    if (isValid) {
+                        return string;
                     }
-                }
-                if (isValid) {
-                    return string;
-                }
 
+                }
             }
         }
-    }
-
-    /**
-     * Method to start the twoPlayerMode
-     */
 
-    public void startGameTp() {
-        int size = 0;
-        Scanner scanner = new Scanner(System.in);
-        while (size < 1 || size > 20) {
-            System.out.println("Choose boardsize (1-20): ");
-            size = scanner.nextInt();
-            if (size < 1 || size > 20) {
-                System.out.println("Invalid input. Choose a size between 1-20.");
+        /**
+         * Method to start the twoPlayerMode
+         */
+        public void startGameTp() {
+            int size = 0;
+            int bombs = 0;
+            Scanner scanner = new Scanner(System.in);
+            while (size < 3 || size > 20) {
+                System.out.println("Choose board size (3-20): ");
+                size = scanner.nextInt();
+                if (size < 3 || size > 20) {
+                    System.out.println("Invalid input. Choose a size between 3-20.");
+                }
+            }
+            while (bombs < 1 || bombs >= (size * size)) {
+                System.out.println("Choose amount of mines (1-" + (size * size - 1) + "): ");
+                bombs = scanner.nextInt();
+                if (bombs < 1 || bombs >= (size * size)) {
+                    System.out.println("Invalid input. Choose a  number between 1-" + (size * size - 1) + ".");
+                }
             }
+            currentPlayer = playerOne;
+            playGameTp();
         }
-        System.out.println("Choose amount of mines: ");
-        int bombs = scanner.nextInt();
-        board = new Board(size, bombs);
```

Comment:
Ska denna rad verkligen bort? Får en krasch när jag försöker starta 2-player för att this.board är null